app/grandchallenge/reader_studies/forms.py

```diff
@@ -688,6 +689,67 @@ class Meta(PermissionRequestUpdateForm.Meta):
         model = ReaderStudyPermissionRequest
 
 
+class GroundTruthViaAnswersForm(SaveFormInitMixin, Form):
+    user = ModelChoiceField(
+        queryset=get_user_model().objects.none(),
+        required=True,
+        help_text="Select a user whose answers will be copied.",
+        widget=Select2Widget,
+    )
+
+    def __init__(self, *args, reader_study, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self._reader_study = reader_study
+
+        self.fields["user"].queryset = (
+            get_user_model()
+            .objects.filter(answer__question__reader_study=reader_study)
+            .distinct()
+        )
+
+    @cached_property
+    def answers(self):
+        return Answer.objects.filter(
+            question__in=self._reader_study.ground_truth_applicable_questions,
+            creator=self.cleaned_data["user"],
+        )
+
+    def clean(self):
+        super().clean()
+
+        if self._reader_study.has_ground_truth:
+            raise ValidationError(
+                "Reader study already has ground truth. Ground truth cannot be updated. "
+                "Please, first delete the ground truth."
+            )
+
+    def clean_user(self):
+        user = self.cleaned_data["user"]
+
+        progress = self._reader_study.get_progress_for_user(user)
+
+        if (
+            self._reader_study.answerable_questions.exists()
+            and progress["diff"] != 0
+        ):
+            raise ValidationError("User has not completed the reader study!")
+
+        return user
+
+    def create_ground_truth(self):
+        answers = self.answers
+
+        for answer in answers:
+            answer.is_ground_truth = True
+
+            answer.pk = None  # Ensures new ones are created
+            answers.id = None  # Required to ensure inheritence works correctly
```

Comment:
`id` is being set on `answers` here and not `answer`. This sets the Python identifier and not the Django `id`.

Reply:
I feel like a dufus for misstyping this. Sorry!